rio/tests/serialization.ml

```diff
@@ -0,0 +1,47 @@
+open Frontend
+open OUnit2
+
+let make_test name filename val_str =
+  name >:: fun _ -> assert_equal val_str (Json.main filename) ~printer:Fun.id
```

Comment:
Can we refactor this a little? Take the file name, and add `"../../../../progs/work_conserving"` and `".sched"` to it here in this function. Then, in the examples below, you will be able to just do, e.g.,:
```
    make_test "single class policy"
      "drop_a_class"
      "{\"fifo\":[{\"class\":\"A\"}]}";
```